service/mapper/transitions_internal_test.go

```diff
@@ -136,771 +123,729 @@ func TestTransitions_BootstrapState(t *testing.T) {
+	t.Run("writeing payloads disabled", func(t *testing.T) {
 		t.Parallel()
 
-		index := mocks.BaselineWriter(t)
-		index.HeaderFunc = func(uint64, *flow.Header) error {
-			return mocks.GenericError
-		}
-
-		tr, st := baselineFSM(t, StatusForwarded)
-		tr.index = index
+		tr, st := baselineFSM(t, StatusCollect)
+		tr.cfg.SkipRegisters = true
 
-		err := tr.IndexChain(st)
+		err := tr.CollectRegisters(st)
 
-		assert.Error(t, err)
+		require.NoError(t, err)
+		assert.Empty(t, st.registers)
+		assert.Equal(t, StatusForward, st.status)
 	})
 
-	t.Run("handles chain failure to retrieve transactions", func(t *testing.T) {
+	t.Run("handles invalid status", func(t *testing.T) {
 		t.Parallel()
 
-		chain := mocks.BaselineChain(t)
-		chain.TransactionsFunc = func(uint64) ([]*flow.TransactionBody, error) {
-			return nil, mocks.GenericError
-		}
-
-		tr, st := baselineFSM(t, StatusForwarded)
-		tr.chain = chain
+		tr, st := baselineFSM(t, StatusBootstrap)
 
-		err := tr.IndexChain(st)
+		err := tr.CollectRegisters(st)
 
 		assert.Error(t, err)
+		assert.Empty(t, st.registers)
 	})
 
-	t.Run("handles chain failure to retrieve transaction results", func(t *testing.T) {
+	t.Run("handles missing tree for commit", func(t *testing.T) {
 		t.Parallel()
 
-		chain := mocks.BaselineChain(t)
-		chain.ResultsFunc = func(uint64) ([]*flow.TransactionResult, error) {
-			return nil, mocks.GenericError
-		}
-
-		tr, st := baselineFSM(t, StatusForwarded)
-		tr.chain = chain
+		tr, st := baselineFSM(t, StatusCollect)
+		st.forest = mocks.BaselineForest(t, false)
 
-		err := tr.IndexChain(st)
+		err := tr.CollectRegisters(st)
 
 		assert.Error(t, err)
+		assert.Empty(t, st.registers)
 	})
+}
 
-	t.Run("handles indexer failure to write transactions", func(t *testing.T) {
+func TestTransitions_MapRegisters(t *testing.T) {
+	t.Run("nominal case with registers to write", func(t *testing.T) {
 		t.Parallel()
 
-		index := mocks.BaselineWriter(t)
-		index.ResultsFunc = func([]*flow.TransactionResult) error {
-			return mocks.GenericError
+		// Path 2 and 4 are the same so the map effectively contains 5 entries.
+		testRegisters := map[ledger.Path]*ledger.Payload{
+			mocks.GenericLedgerPath(0): mocks.GenericLedgerPayload(0),
+			mocks.GenericLedgerPath(1): mocks.GenericLedgerPayload(1),
+			mocks.GenericLedgerPath(2): mocks.GenericLedgerPayload(2),
+			mocks.GenericLedgerPath(3): mocks.GenericLedgerPayload(3),
+			mocks.GenericLedgerPath(4): mocks.GenericLedgerPayload(4),
+			mocks.GenericLedgerPath(5): mocks.GenericLedgerPayload(5),
 		}
 
-		tr, st := baselineFSM(t, StatusForwarded)
-		tr.index = index
+		write := mocks.BaselineWriter(t)
+		write.PayloadsFunc = func(height uint64, paths []ledger.Path, value []*ledger.Payload) error {
+			assert.Equal(t, mocks.GenericHeight, height)
 
-		err := tr.IndexChain(st)
+			// Expect the 5 entries from the map.
+			assert.Len(t, paths, 6)
+			assert.Len(t, value, 6)
```

Comment:
Why do we expect 5 but assert 6? 🤔 The diff is a bit broken so I don't see what this used to be.